app/bundles/LeadBundle/EventListener/SegmentFilterTypeaheadSubscriber.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Mautic\LeadBundle\EventListener;
+
+use Mautic\CoreBundle\Factory\ModelFactory;
+use Mautic\LeadBundle\Event\ListTypeaheadEvent;
+use Mautic\LeadBundle\Model\CompanyModel;
+use Mautic\LeadBundle\Model\FieldModel;
+use Mautic\LeadBundle\Model\LeadModel;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class SegmentFilterTypeaheadSubscriber implements EventSubscriberInterface
+{
+    public function __construct(private ModelFactory $modelFactory)
```

Comment:
I'd understand to use ModelFactory to get models dynamically based on some entity name. But here you just need 3 models. Please add them into the constructor instead of the ModelFactory. Let's be explicit of what a service needs for dependencies.